src/auth/authOS.ts

```diff
@@ -0,0 +1,193 @@
+import {useMutation, UseMutationOptions, useQuery, UseQueryOptions} from 'react-query'
+
+import {
+  AUTH_METHOD_KEY,
+  AUTH_METHOD_PIN,
+  ENCRYPTED_PIN_HASH_KEY,
+  useMutationWithInvalidations,
+  useWallet,
+} from '../hooks'
+import {WalletMeta} from '../legacy/state'
+import storage from '../legacy/storage'
+import {Storage} from '../Storage'
+import {WalletJSON, walletManager, YoroiWallet} from '../yoroi-wallets'
+import {AuthenticationPrompt, authOsEnabledOnDevice, KeychainStorage} from './KeychainStorage'
+import {AuthMethod} from './types'
+
+export const useAuthOsEnabledOnDevice = (options?: UseQueryOptions<boolean, Error>) => {
+  const query = useQuery({
+    cacheTime: 0,
+    queryKey: ['useAuthOsEnabledOnDevice'],
+    queryFn: authOsEnabledOnDevice,
+    suspense: true,
+    ...options,
+  })
+
+  // use refetch on app settings
+  return {
+    ...query,
+    authOsEnabledOnDevice: query.data,
+  }
+}
+
+export const AUTH_METHOD_OS: AuthMethod = 'os'
+export const useEnableAuthWithOs = (
+  {authenticationPrompt, storage}: {authenticationPrompt: AuthenticationPrompt; storage: Storage},
+  options?: UseMutationOptions<void, Error>,
+) => {
+  const mutation = useMutationWithInvalidations({
+    mutationFn: () =>
+      KeychainStorage.initializeAppAuth()
+        .then(() => KeychainStorage.appAuth(authenticationPrompt))
+        .then(() => storage.setItem(AUTH_METHOD_KEY, JSON.stringify(AUTH_METHOD_OS)))
+        // current behavior - on auth with os it deletes the pin stored
```

Comment:
"current" is a really problematic word
does this mean the code im looking at right now? or the code on develop before this pr is merged?
time and order are dependencies that we should strive to eliminate

Reply:
stick to defining rules and relationships between data
no time or order required